SwiftCompilerSources/Sources/Optimizer/Utilities/Test.swift

```diff
@@ -172,47 +207,37 @@ public func registerSILTests() {
 //   - its type
 //   - something to identify the instance (mostly this means calling dump)
 let parseTestSpecificationTest = 
-FunctionTest(name: "test_specification_parsing") { function, arguments, context in 
-  struct _Stderr : TextOutputStream {
-    public init() {}
-
-    public mutating func write(_ string: String) {
-      for c in string.utf8 {
-        _swift_stdlib_putc_stderr(CInt(c))
-      }
-    }
-  }
-  var stderr = _Stderr()
+FunctionTest("test_specification_parsing") { function, arguments, context in 
   let expectedFields = arguments.takeString()
   for expectedField in expectedFields.string {
     switch expectedField {
     case "A":
       let argument = arguments.takeArgument()
-      print("argument:\n\(argument)", to: &stderr)
+      print("argument:\n\(argument)", to: &standardError)
```

Comment:
Why do we want to print test results to stderr and not to stdout?

If there is a good reason for this, you should add a dedicated print function for test results.

Reply:
It's a bit more convenient to use `dump()` than `print(std::outs())` on the C++ side.  And both sides need to write to the same file and flush before switching to the other language's handle.  Anyway, here's a PR https://github.com/apple/swift/pull/69080 to use stdout.

Reply:
We definitely need to be able to write debug output to stderr. So `standardError` needs to be available for debugging. But I see no problem with the unit tests printing to stdout.